cli/context/store/tlsstore.go

```diff
@@ -15,29 +15,20 @@ func (s *tlsStore) contextDir(id contextdir) string {
 	return filepath.Join(s.root, string(id))
 }
 
-func (s *tlsStore) endpointDir(contextID contextdir, name string) string {
-	return filepath.Join(s.root, string(contextID), name)
-}
-
-func (s *tlsStore) filePath(contextID contextdir, endpointName, filename string) string {
```

Comment:
I'm slightly hesitant on removing these. IMO there's some value in having these helper functions due to the signature prompting you on what you should supply to obtain the final directory.
If I would be writing a new code and need to obtain the file path, then I would have to search for other code that does this to know how to construct the whole path with `filepath.Join`.
With those conversion functions I just need to know that there is this function and then the arguments to the functions will guide me on what I need to provide. For example if I miss the `endpointName` then the compiler will yell at me, whereas if I forget it in `filepath.Join` then I won't know that at the compile time.
WDYT?

Reply:
Let me have a look. My main reasons were that;

- Each of them would be reconstructing the parent directory, so I initially wanted to re-use `contextDir()` and `endpointDir()`
- The `fileDir()` one felt really redundant
- The `endpointDir()` as used in `listContextData()` was really silly, as we already were traversing the directory, and then were re-constructing the name … to get the name of the directory we were in.

Perhaps I can bring back `endpointDir()`, and only remove `filePath()`; let me see how that looks.

There were some other changes I was considering;

- As the `contextDir` depends on the _context_ store, I want to look at passing that directory in (effectively, calculating that path should be something done by the _context_ store)
- Given that for the TLS-store we _only_ want `ca.pem`, `cert.pem` and `key.pem`, perhaps add accessors for those instead of trying to keep it “too generic”

A lot of the code was written to be really “generic”, but that’s not how it’s used, and perhaps we should rewrite more code to be specific to how it’s _actually_ used.

Reply:
Makes sense to me, thanks for the explanation! 👍